From 4dd456157992001fddbaf10c2fc46c59efc4f2f0 Mon Sep 17 00:00:00 2001
From: Doug Ledford <dledford@redhat.com>
Date: Wed, 24 Sep 2014 14:58:20 -0400
Subject: [infiniband] usnic: Fix endianness-related warnings

Message-id: <d10b32da80378d23e23e7353e5a2acbf493dde2f.1411569661.git.dledford@redhat.com>
Patchwork-id: 96598
O-Subject: [Patch RHEL7 18/29] IB/usnic: Fix endianness-related warnings
Bugzilla: 916382
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Upinder Malhi <umalhi@cisco.com>

Fix sparse endianness related warnings.

Signed-off-by: Upinder Malhi <umalhi@cisco.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 4942c0b4b64478ff45c3bbf4d40aebd66de0bcc5)
Signed-off-by: Doug Ledford <dledford@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/usnic/usnic_common_util.h b/drivers/infiniband/hw/usnic/usnic_common_util.h
index afd8bfa..9d737ed 100644
--- a/drivers/infiniband/hw/usnic/usnic_common_util.h
+++ b/drivers/infiniband/hw/usnic/usnic_common_util.h
@@ -36,7 +36,7 @@ usnic_mac_to_gid(const char *const mac, char *raw_gid)
 }
 
 static inline void
-usnic_mac_ip_to_gid(const char *const mac, const uint32_t inaddr, char *raw_gid)
+usnic_mac_ip_to_gid(const char *const mac, const __be32 inaddr, char *raw_gid)
 {
  raw_gid[0] = 0xfe;
  raw_gid[1] = 0x80;
diff --git a/drivers/infiniband/hw/usnic/usnic_transport.c b/drivers/infiniband/hw/usnic/usnic_transport.c
index 9182cc8..ddef6f7 100644
--- a/drivers/infiniband/hw/usnic/usnic_transport.c
+++ b/drivers/infiniband/hw/usnic/usnic_transport.c
@@ -59,8 +59,7 @@ int usnic_transport_sock_to_str(char *buf, int buf_sz,
  if (err)
   return 0;
 
- addr = htonl(addr);
- return scnprintf(buf, buf_sz, "Proto:%u Addr:%pI4 Port:%hu",
+ return scnprintf(buf, buf_sz, "Proto:%u Addr:%pI4h Port:%hu",
    proto, &addr, port);
 }
 
-- 
1.7.1